test/e2e_node/container_lifecycle_test.go

```diff
@@ -3568,7 +4289,415 @@ var _ = SIGDescribe(nodefeature.SidecarContainers, "Containers Lifecycle", func(
+
+			})
+		})
+	})
+
+	ginkgo.When("A restartable init container has its image updated during pod termination", func() {
+		init1 := "init-1"
+		restartableInit2 := "restartable-init-2"
+		restartableInit3 := "restartable-init-3"
+		regular1 := "regular-1"
+
+		podTerminationGracePeriodSeconds := int64(180)
+		containerTerminationSeconds := 10
+
+		updatedImage := busyboxImage
+
+		var originalPodSpec *v1.Pod
+
+		ginkgo.BeforeEach(func(ctx context.Context) {
+			originalPodSpec = &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "terminate-restartable-init-gracefully-with-img-update",
+				},
+				Spec: v1.PodSpec{
+					TerminationGracePeriodSeconds: &podTerminationGracePeriodSeconds,
+					RestartPolicy:                 v1.RestartPolicyAlways,
+					InitContainers: []v1.Container{
+						{
+							Name:  init1,
+							Image: agnhostImage,
+							Command: ExecCommand(init1, execCommand{
+								Delay:              1,
+								TerminationSeconds: 5,
+								ExitCode:           0,
+							}),
+						},
+						{
+							Name:  restartableInit2,
+							Image: agnhostImage,
+							Command: ExecCommand(restartableInit2, execCommand{
+								Delay:              600,
+								TerminationSeconds: containerTerminationSeconds,
+								ExitCode:           0,
+							}),
+							RestartPolicy: &containerRestartPolicyAlways,
+						},
+						{
+							Name:  restartableInit3,
+							Image: agnhostImage,
+							Command: ExecCommand(restartableInit3, execCommand{
+								Delay:              600,
+								TerminationSeconds: containerTerminationSeconds,
+								ExitCode:           0,
+							}),
+							RestartPolicy: &containerRestartPolicyAlways,
+						},
+					},
+					Containers: []v1.Container{
+						{
+							Name:  regular1,
+							Image: agnhostImage,
+							Command: ExecCommand(regular1, execCommand{
+								Delay:              600,
+								TerminationSeconds: containerTerminationSeconds,
+								ExitCode:           0,
+							}),
+						},
+					},
+				},
+			}
+		})
+
+		restartableInitContainerGracefulTerminationImageUpdateTest := func(ctx context.Context) {
+			preparePod(originalPodSpec)
+
+			client := e2epod.NewPodClient(f)
+			pod := client.Create(ctx, originalPodSpec)
+
+			ginkgo.By("Running the pod")
+			err := e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Deleting the pod")
+			err = client.Delete(ctx, pod.Name, metav1.DeleteOptions{GracePeriodSeconds: &podTerminationGracePeriodSeconds})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Updating the image")
+			client.Update(ctx, pod.Name, func(pod *v1.Pod) {
+				pod.Spec.InitContainers[1].Image = updatedImage
+			})
+
+			//TODO verify this is the intended behavior
+			//during termination of the regular and last restartable init container
+			ginkgo.By("ensuring the restartable init container does not restart during termination", func() {
```

Comment:
This seems like a corner case but the sidecar likely should not restart while the pod is terminating

Reply:
This looks being fixed in the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/4438-container-restart-termination/README.md) though the case where an image is updated is not described explicitly.

It might be better to add a comment like this:
https://github.com/kubernetes/kubernetes/blob/a6ea7b8218d01b504fda2ef4988627fe4194fcaa/test/e2e_node/container_lifecycle_test.go#L942-L945